remotespark/kernels/kernelmagics.py

```diff
@@ -264,9 +285,12 @@ def _do_not_call_start_session(self, line, cell="", local_ns=None):
         return self.session_started
 
     @cell_magic
+    @handle_expected_exceptions
     def _do_not_call_delete_session(self, line, cell="", local_ns=None):
-        if self.session_started:
-            self.spark_controller.delete_session_by_name(self.session_name)
+        try:
+            if self.session_started:
+                self.spark_controller.delete_session_by_name(self.session_name)
+        finally:
```

Comment:
Maybe add this code so that any other developer reading it doesn't wonder the same thing?

```
except:
    # Exception is already caught internally, logged, and rethrown
    pass
```